### PR TITLE
refactor gadgettracermanager and add unit tests

### PR DESCRIPTION
**Pull request by alban**
GadgetTracerManager is refactored so it's possible to create an instance
that will not do any privileged operations like using BPF maps.

Then, it's possible to add unit tests.

### CODE REVIEW

File: pkg/gadgettracermanager/gadgettracermanager_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+					&pb.Label{Key: "key2", Value: "value2"},
+				},
+			},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "value2"},
```
Comment (mauriciovasquezbernal) [inline pkg/gadgettracermanager/gadgettracermanager_test.go:61]: We can add a third label and it still will match, right? (just to check that the matching algorithm matches a subset of labels)

File: pkg/gadgettracermanager/gadgettracermanager_test.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+			match:       true,
+			selector:    &pb.ContainerSelector{},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+			},
+		},
+		{
+			description: "Selector without all filters",
```
Comment (mauriciovasquezbernal) [inline pkg/gadgettracermanager/gadgettracermanager_test.go:44]: Which filter is missing?

File: pkg/gadgettracermanager/gadgettracermanager_test.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+		t.Fatalf("Error while checking tracers: len %d", len(g.tracers))
+	}
+	_, ok := g.tracers["my_trace_id0"]
+	if !ok {
+		t.Fatalf("Error while checking tracer %s: not found", "my_trace_id0")
+	}
+	_, ok = g.tracers["my_trace_id2"]
+	if !ok {
+		t.Fatalf("Error while checking tracer %s: not found", "my_trace_id2")
+	}
```
Comment (mauriciovasquezbernal) [inline pkg/gadgettracermanager/gadgettracermanager_test.go:161]: Some ideas for more test cases:
- try to add an already existing tracer
- remove a non existing tracer

File: pkg/gadgettracermanager/gadgettracermanager_test.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	_, ok := g.containers["abcde0"]
+	if !ok {
+		t.Fatalf("Error while checking container %s: not found", "abcde0")
+	}
+	_, ok = g.containers["abcde2"]
+	if !ok {
+		t.Fatalf("Error while checking container %s: not found", "abcde2")
+	}
+
```
Comment (mauriciovasquezbernal) [inline pkg/gadgettracermanager/gadgettracermanager_test.go:212]: Same ideas as above.

**Review by mauriciovasquezbernal — CHANGES_REQUESTED** [review]
Very nice change! I think later on we can create more tests that actually check that the eBPF maps are created and populated correctly. 

I have a blocking comment on a conditional that is misplaced, otherwise LGTM!

File: pkg/gadgettracermanager/gadgettracermanager_test.go (hunk 2 above)
Comment (alban) [reply pkg/gadgettracermanager/gadgettracermanager_test.go:44]: Oops, I meant

```suggestion
			description: "Selector with all filters",
```

**Comment by alban** [comment]
Fixed. PTAL.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Thanks for handling all the comments.

**Pull request merged**